scalafmt-core/shared/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -676,7 +679,13 @@ object TreeOps {
     }
 
   // Redundant {} block around case statements
-  def isCaseBodyABlock(ft: FormatToken, caseStat: Case): Boolean =
-    ft.right.is[Token.LeftBrace] && (caseStat.body eq ft.meta.rightOwner)
+  def isCaseBodyABlock(ft: FormatToken, caseStat: Tree): Boolean = {
+    val body = caseStat match {
+      case tc: TypeCase => tc.body
+      case c: Case => c.body
```

Comment:
this will throw if parameter is neither

Reply:
Added an option and just return false in that case.